system/lib/libc/dynlink.c

```diff
@@ -223,23 +266,25 @@ static void ensure_init() {
 void* dlopen(const char* file, int flags) {
   ensure_init();
   if (!file) {
-    return head;
+    return head->dso;
```

Comment:
What if this is a dlsym event and not another dlopen?

Reply:
`head` is always the primary `__main__` dso.  Its ensured by `ensure_init`